src/python/dxpy/bindings/dxfile_functions.py

```diff
@@ -466,3 +466,50 @@ def upload_string(to_upload, media_type=None, keep_open=False, wait_on_close=Fal
         handler.close(block=wait_on_close, **remaining_kwargs)
 
     return handler
+
+def download_folder(project, destdir, folder="/", chunksize=dxfile.DEFAULT_BUFFER_SIZE):
```

Comment:
Hello @psung ,

thank you for the idea! I hope I managed to port a **dx download -r** command to use **dxpy.download_folder()** API call. To do that I introduced a **dxpy.list_subfolders()** function, which provides a directory listings cache support and which is called either by **dxpy.download_folder()** function or by **dx** CLI command (see [here](https://github.com/storozhilov-dnanexus/dx-toolkit/blob/PTFM-17466-python-download_folder-function/src/python/dxpy/cli/download.py#L134)).

But I have a little concern about the respective [TravisCI build](https://travis-ci.org/storozhilov-dnanexus/dx-toolkit/builds/137848025), which failed for each configuration with following error:

```
...
creating build/bdist.linux-x86_64/wheel/dxpy-c0b0e43.dist-info/WHEEL
unset PYTHONPATH; source "/home/travis/build/storozhilov-dnanexus/dx-toolkit/src/../build/py_env/bin/activate"; pip install --ignore-installed --prefix="/home/travis/build/storozhilov-dnanexus/dx-toolkit/src/.." python/dist/*.whl
dxpy-c0b0e43-py2.py3-none-any.whl is not a valid wheel filename.
make[1]: *** [/home/travis/build/storozhilov-dnanexus/dx-toolkit/src/../bin/dx] Error 1
make[1]: Leaving directory `/home/travis/build/storozhilov-dnanexus/dx-toolkit/src'
```

That looks strange, because Travis build runs fine on my workstation, so I can not reproduce that locally to be able to investigate. Maybe you can help somehow?

Thank you and have a nice day,
Ilya